src/crypto.rs

```diff
@@ -556,6 +569,58 @@ impl TryFrom<&serde_cbor::Value> for COSEKey {
     }
 }
 
+impl TryFrom<&X509PublicKey> for COSEKey {
+    type Error = WebauthnError;
+    fn try_from(cert: &X509PublicKey) -> Result<COSEKey, Self::Error> {
+        let ec_key = cert
+            .pubk
+            .public_key()
+            .and_then(|pk| pk.ec_key())
+            .map_err(WebauthnError::OpenSSLError)?;
+        let ec_group = ec_key.group();
+
+        let key = match cert.t {
+            COSEContentType::ECDSA_SHA256
+            | COSEContentType::ECDSA_SHA384
+            | COSEContentType::ECDSA_SHA512 => {
+                let curve = ec_group
+                    .curve_name()
+                    .ok_or(WebauthnError::OpenSSLErrorNoCurveName)
+                    .and_then(ECDSACurve::try_from)?;
+
+                let mut ctx =
+                    openssl::bn::BigNumContext::new().map_err(WebauthnError::OpenSSLError)?;
+                let mut x = openssl::bn::BigNum::new().map_err(WebauthnError::OpenSSLError)?;
+                let mut y = openssl::bn::BigNum::new().map_err(WebauthnError::OpenSSLError)?;
+
+                ec_group
+                    .generator()
```

Comment:
I think this may be your issue, because you are generating a new curve here, not extracting the x/y from ec_key. You need to change this so it's:

```
let ec_pointref = ec_key.public_key.map_err()?
ec_pointref.affine_coordinates_gfp(...)
```

Reply:
Ah—this seems to fix it. I didn't catch `public_key` as a method that returned an `ECPointRef` and `generator` was the only one I could find.

Reply:
It's in the deref section of the docs, slightly further down. Anyway, happy this helped